src/guest_memory.rs

```diff
@@ -246,6 +246,22 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     unsafe fn as_mut_slice(&self) -> Option<&mut [u8]> {
         None
     }
+
+    /// Returns a [`VolatileSlice`](struct.VolatileSlice.html) of `count` bytes starting at
+    /// `offset`.
+    fn get_slice(
+        &self,
+        _offset: MemoryRegionAddress,
+        _count: usize,
+    ) -> Result<volatile_memory::VolatileSlice> {
+        Err(Error::HostAddressNotAvailable)
+    }
+
+    /// Gets a slice of memory for the entire region that supports volatile access.
+    fn as_volatile_slice(&self) -> volatile_memory::VolatileSlice {
```

Comment:
Please also add a doc comment to explain how you can use this functionality and why. Otherwise people might just miss it. Same for the other functions you are adding.

If it's not an overkill, I would also recommend adding an example.

Reply:
I would actually recommend adding _only_ an example, for example creating a VolatileRef<u32> from the VolatileSlice and storing 42 into it.

Reply:
Cool! I will write this example to reveal the world's top secret.